app.js

```diff
@@ -27,61 +27,53 @@ function generateStates(cards, cardback, predicate, sideways, transform)
   , "DeckIDs": []
   };
 
-  var current_id = 100;
-  for(cardix in cards)
-  {
-    card = cards[cardix];
+  cards.forEach(function (card) {
+    var card_info = card["data"];
     for(var i = 0; i < card["amount"]; i++)
     {
-      card_info = db["cards"][card["id"]];
-      if(!predicate(card_info))
-        continue;
       if(obj_state["CustomDeck"].hasOwnProperty(card_info["pic_id"]))
         var deck_obj = obj_state["CustomDeck"][card_info["pic_id"]];
       else
       {
         var deck_obj = idToDeckObj(card_info["pic_id"], cardback);
         obj_state["CustomDeck"][card_info["pic_id"]] = deck_obj;
       }
-      card_id = parseInt(card_info["pic_id"] + ("0" + card_info["index"].toString()).slice(-2));
+      var card_id = parseInt(card_info["pic_id"] + ("0" + card_info["index"].toString()).slice(-2));
       obj_state["DeckIDs"].push(card_id);
-      deck_obj_ = {};
+      var deck_obj_ = {};
       deck_obj_[card_info["pic_id"]] = deck_obj;
       var card_obj =
       { "CardID": card_id
       , "Name": "Card"
       , "Nickname": card_info["name"]
       , "Transform": transform
       , "CustomDeck": deck_obj_
-      , "SidewaysCard": card_info["type"] === "Problem"
+      , "SidewaysCard": sideways
```

Comment:
The sideways argument in this function is a property of the entire deck object, here however we set whether an individual card in the deck is sideways or not. While with the current implementation, only the Problem deck is sideways (meaning that it will have only sideways cards), but generally not every card can be sideways in a sideways deck and some cards may be sideways in a regular deck, so I'd rather leave this as is.

Reply:
Maybe in some games (and it took me longer than it should to remember MTG's split cards), but in MLP being a problem is equivalent to being in the problem deck. Or were you planning to make dilemmas sideways?

Reply:
No, I'm not, but who knows what might or might not change in the future. The original code is more generalized, so I prefer it.